File: common-lib/src/main/java/org/bank/dto/response/BillResponseDTO.java
package org.bank.dto.response;

import com.fasterxml.jackson.annotation.JsonFormat;

import java.math.BigDecimal;
import java.time.OffsetDateTime;

public record BillResponseDTO(
        Long billId,
        Long accountId,
        BigDecimal amount,
        Boolean isDefault,
        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssXXX")
        OffsetDateTime creationDate,
        Boolean overdraftEnabled
) {}

File: frontend/src/dto/response/bill-response.dto.ts
export interface BillResponseDTO {
  billId: number
  accountId: number
  amount: number
  isDefault: boolean
  creationDate: string
  overdraftEnabled: boolean
}

export interface BillDepositResponseDTO {
  billId: number
  accountId: number
  amount: number
  email: string
  isDefault: boolean
  overdraftEnabled: boolean
  creationDate: string
}
